services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/northbound/flows/AutoRerouteSpec.groovy

```diff
@@ -120,4 +123,54 @@ class AutoRerouteSpec extends BaseSpecification {
         flowPath?.first() && northboundService.portUp(flowPath.first().switchId, flowPath.first().portNo)
         Wrappers.wait(5) { northboundService.getAllLinks().every { it.state != IslChangeType.FAILED } }
     }
+
+    def "Flow in 'Up' status doesn't try to reroute even though more preferable path is available"() {
+        given: "Two active switches and a flow with one alternate path at least"
+        def switches = topology.getActiveSwitches()
+        List<List<PathNode>> possibleFlowPaths = []
+        def (Switch srcSwitch, Switch dstSwitch) = [switches, switches].combinations()
+                .findAll { src, dst -> src != dst }.find { Switch src, Switch dst ->
+            possibleFlowPaths = topologyEngineService.getPaths(src.dpId, dst.dpId)*.path.sort { it.size() }
+            possibleFlowPaths.size() > 1
+        }
+        assumeTrue("No suiting switches found", srcSwitch && dstSwitch)
+
+        def flow = flowHelper.randomFlow(srcSwitch, dstSwitch)
+        flow.maximumBandwidth = 1000
+        northboundService.addFlow(flow)
+        def flowPath = PathHelper.convert(northboundService.getFlowPath(flow.id))
+        assert Wrappers.wait(5) { northboundService.getFlowStatus(flow.id).status == FlowState.UP }
+
+        and: "Make current flow path less preferable than others"
+        pathHelper.makePathMorePreferable(possibleFlowPaths.find { it != flowPath }, flowPath)
+
+        when: "One of the links not used by flow goes down"
+        def involvedIsls = pathHelper.getInvolvedIsls(flowPath)
+        def islToFail = topology.islsForActiveSwitches.find { !involvedIsls.contains(it) }
+        northboundService.portDown(islToFail.srcSwitch.dpId, islToFail.srcPort)
+        northboundService.portDown(islToFail.dstSwitch.dpId, islToFail.dstPort)
+
+        then: "Link status becomes 'FAILED' and flow status is 'Up'"
```

Comment:
Don't think that it makes sense to assert flow status here. If you still want to do that, it makes no sense without waiting a rerouteTimeout seconds (since anyway flow will not change its status immediately)